src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4714,5 +4705,377 @@ .maxstack  1
 }
 ");
         }
+
+        [Fact]
+        public void LocalScope_DeclarationExpression_01()
+        {
+            var source = """
+                ref struct RS
+                {
+                    public RS(ref RS rs) => throw null!;
+                }
+
+                class Program
+                {
+                    static void M0(ref RS rs1, out RS rs2)
+                    {
+                        // ok. RSTE of rs1 is ReturnOnly. STE of rs2 is ReturnOnly.
+                        rs2 = new RS(ref rs1);
+                    }
+
+                    static RS M1(scoped ref RS rs3)
+                    {
+                        // RSTE of rs3 is CurrentMethod
+                        // STE of rs4 (local variable) is also CurrentMethod
+                        M0(ref rs3, out var rs4);
+                        return rs4; // 1
+                    }
+
+                    static RS M2(scoped ref RS rs3)
+                    {
+                        M0(ref rs3, out RS rs4);
+                        return rs4; // 2
+                    }
+
+                    static RS M3(scoped ref RS rs3)
+                    {
+                        RS rs4;
+                        M0(rs3, out rs4); // 3
```

Comment:
>M0(rs3, [](http://example.com/codeflow?start=24&length=7)

Should be `ref rs3`. (Sorry, my code suggestion above was incorrect.)

Reply:
Oops 😓 I should have spotted that the resulting diagnostic did not look useful..